src/instafeed.js

```diff
@@ -117,6 +119,9 @@
       url = 'https://graph.instagram.com/me/media?fields=caption,id,media_type,media_url,permalink,thumbnail_url,timestamp,username&access_token='+ token;
       scope._debug('onTokenReceived', 'request url', url);
 
+      //Call the before hook
+      scope._runHook('before');
```

Comment:
So, I'm thinking we might want to run this hook before the call to `getAccessToken`, rather than after.

I'm thinking we take what is currently L95, where set the `running` flag, and turn it into a new method called `_start()`:

https://github.com/stevenschobert/instafeed.js/blob/4dfedbe0bdcdeda0055dffbdf2c1e3ae9dbea8f4/src/instafeed.js#L94-L95

The `_start()` method could look pretty much like `_finish()` method, except it would set the `running` flag to true first, then call the hook.

https://github.com/stevenschobert/instafeed.js/blob/4dfedbe0bdcdeda0055dffbdf2c1e3ae9dbea8f4/src/instafeed.js#L358-L361

Reply:
Excellent points all. Changes coming up!